controllers/jobs.js

```diff
@@ -4,24 +4,67 @@ const {BadRequestError, NotFoundError} = require('../errors');
 
 // Controller to handle getting all jobs
 const getAllJobs = async (req, res) => {
-    res.send('get all jobs');// response
+   const jobs = await Job.find({createdBy: req.user.userId}).sort('createdAt');
+   res.status(StatusCodes.OK).json({jobs, count: jobs.length});
 }
+
 // Controller to handle getting a single job
 const getJob = async (req, res) => {
-    res.send('get job');
+    const {
+        user:{userId}, 
+        params:{id:jobId}
+    } = req;
+
+    const job = await Job.findOne({
```

Comment:
Nitpick (deson't require changes):
I'd recommend checking if `req.params.id` or **any other form of stringified id** is a valid ObjectID by using mongoose's [isValidObjectId utility function](https://mongoosejs.com/docs/5.x/docs/api.html#mongoose_Mongoose-isValidObjectId) BEFORE using it within a `Job` method. Otherwise, if you supply an invalid id, then... for example, `Job.findOne` may throw a casting error:
```
CastError: Cast to ObjectId failed for value "1" (type string) at path "_id" for model "Job"
```
This should also apply to your other controllers that use an `id` to find, delete, or update a `Job`.
—-
For example:
```js
const mongoose = require(“mongoose”);
…

  if (!mongoose.isValidObjectId(jobId)) {
    throw new BadRequestError("Invalid job id");
  }

  const job = await Job.findOne({ _id: jobId, createdBy: userId });

  …etc

```